service/src/main/kotlin/app/cash/backfila/service/runner/statemachine/BatchQueuer.kt

```diff
@@ -60,8 +66,11 @@ class BatchQueuer(
       }
 
       try {
+        // The client is expected to compute as many batches as possible up to computeCountLimit
+        // and within computeTimeLimitMs. It may exceed the time limit when computing its last returned batch.
         val computeTimeLimitMs = 5_000L // half of HTTP timeout
         val computeCountLimit = nextBatchChannel.capacity
+          .coerceAtLeast(backfillRunner.factory.backfilaConfig.minimum_batches_per_get_next_batch_call)
```

Comment:
How will this play with coroutines?

Reply:
capacity of this channel is directly tied to the number of threads for the partition

if threads=1, we're only going to get 2 (2x1 the default multiplier) batches per call. I think if we set a minimum of 10 it would reduce calls without negatively impacting anything

if we have an excess, ie more than we can send to the channel, it just blocks. that's already how it works, no problems, coroutines are awesome in that way